core/types/header_ext_test.go

```diff
@@ -242,3 +242,55 @@ func TestHeaderSerializable_updates(t *testing.T) {
 		assert.Equal(t, wantedExtras, extras)
 	})
 }
+
+func TestHeaderExtraPostCopy(t *testing.T) {
+	t.Parallel()
+
+	t.Run("empty_extra", func(t *testing.T) {
+		t.Parallel()
+
+		got := &ethtypes.Header{
+			ParentHash: common.Hash{1},
+		}
+		extra := &HeaderExtra{}
+
+		extra.PostCopy(got)
```

Comment:
```suggestion
		got := CopyHeader(&ethtypes.Header{
			ParentHash: common.Hash{1},
		})
```

Not only does this simplify the test but it also ensures proper integration for the desired behaviour.

Reply:
I just tested locally and you'll also need to add in zero values for `Difficulty` and `Number` in the `want` because they'll be copied.

```go
			Difficulty: big.NewInt(0),
			Number:     big.NewInt(0),
```

Same applies to the other change I recommended for the test.

Reply:
In this case there is already `TestCopyHeader` which pretty much does the same and more, should we then remove this test?

Reply:
It doesn't test that the `HeaderExtra` is empty so I think it's worth keeping.

Reply:
It does, given libevm's CopyHeader does call PostCopy or am I missing something?
Maybe a bit risky to rely on CopyHeader keeping PostCopy in the future though 🤔

Reply:
Oh I thought you meant an upstream test of `CopyHeader`, not the one in `block_ext_test.go`. Yes, that's sufficient.

Reply:
Test removed in favor of `TestCopyHeader`